include/seqan3/alphabet/adaptation/char.hpp

```diff
@@ -94,10 +94,10 @@ struct alphabet<char_type>
      * \param[in,out] chr The alphabet letter that you wish to assign to.
      * \returns A reference to the alphabet letter you passed in.
      */
-    static constexpr char_type & assign_rank_to(decltype(alphabet::to_rank(char_type{})) const rank,
-                                                char_type & chr) noexcept
+    static constexpr char_type & assign_rank_to(size_t const rank, char_type & chr) noexcept
```

Comment:
Did something break when not doing this?

Reply:
yes, but not visually in a test: when I assign a rank that is larger than can be modelled by rank_type the value is implicitly cast and the assert would not catch the misuse.

Reply:
This is not generic, rank types can be signed as well. Also we do not specify in the concept that the validity of the rank is asserted so we cannot enforce this on every alphabet here. Some alphabet may handle out-of-bounds ranks differently.

Reply:
I asked you this on Gitter :(
You actually had an out commented test in the alphabet test template that said `todo assert death when rank is > alphabet size` 
That's why I came across this issue. How should we handle this then?

Reply:
> I asked you this on Gitter :(

Well, I also said I am not available 24h/day.

> todo assert death when rank is > alphabet size

That would only work in debug anyway. It's not something sensible to test. At most it should be tested directly on the member function and not the free function.

> That's why I came across this issue. How should we handle this then?

I am not yet sure it is an issue. Assigning a value of 16543 to a dna4 letter is undefined behaviour. If cast to `uint8_t` makes this resolve to a value that is accepted that is not a problem per se. If we want to catch that we need to change all the assignment functions to handle the largest possible type, assert on that and then always downcast. That costs performance even in NDEBUG.

Reply:
alright, then I'll remove the test and the changes that correspond to this.